game/definitions.rpy

```diff
@@ -794,15 +833,142 @@ init python in utils:
+
+    def jn_is_time_block_early_morning():
+        """
+        Returns True if the current time is judged to be early morning.
+        """
+        return jn_get_current_hour() in range(3, 5)
+
+    def jn_is_time_block_mid_morning():
+        """
+        Returns True if the current time is judged to be mid morning.
+        """
+        return jn_get_current_hour() in range(5, 9)
+
+    def jn_is_time_block_late_morning():
+        """
+        Returns True if the current time is judged to be late morning.
+        """
+        return jn_get_current_hour() in range(9, 12)
+
+    def jn_is_time_block_afternoon():
+        """
+        Returns True if the current time is judged to be afternoon.
+        """
+        return jn_get_current_hour() in range(12, 18)
+
+    def jn_is_time_block_evening():
+        """
+        Returns True if the current time is judged to be evening.
+        """
+        return jn_get_current_hour() in range(18, 22)
+
+    def jn_is_time_block_night():
+        """
+        Returns True if the current time is judged to be night.
+        """
+        return jn_get_current_hour() in range(22, 3)
+
+    def jn_get_holiday_for_date(input_date=None):
+        """
+        Gets the holiday - if any - corresponding to the supplied date, or the current date by default.
+
+        IN:
+            - date - datetime object to test against. Defaults to the current date.
+
+        OUT:
+            - JNHoliday representing the holiday for the supplied date.
+        """
+
+        if input_date is None:
+            input_date = datetime.datetime.today()
+
+        elif not isinstance(input_date, datetime.date):
+            raise TypeError("input_date for holiday check must be of type date; type given was {0}".format(type(input_date)))
+
+        if input_date == JN_NEW_YEARS_DAY:
+            return JNHolidays.new_years_day
+
+        elif input_date == JN_EASTER:
+            return JNHolidays.easter
+
+        elif input_date == JN_HALLOWEEN:
+            return JNHolidays.halloween
+
+        elif input_date == JN_CHRISTMAS_EVE:
+            return JNHolidays.christmas_eve
+
+        elif input_date == JN_CHRISTMAS_DAY:
+            return JNHolidays.christmas_day
+
+        elif input_date == JN_NEW_YEARS_EVE:
+            return JNHolidays.new_years_eve
+
+        else:
+            return JNHolidays.none
+
+init python in utils:
+    # Key setup
+    key_path = os.path.join(renpy.config.basedir, "game/dev/key.txt").replace("\\", "/")
+    if not os.path.exists(key_path):
+        __KEY_VALID = False
+
+    else:
+        with open(name=key_path, mode="r") as key_file:
+            if hashlib.sha256(key_file.read().encode("utf-8")).hexdigest() == __KEY_HASH:
+                __KEY_VALID = True
+
+            else:
+                __KEY_VALID = False
+
+    def jn_get_key_valid():
+        """
+        Returns the validation state of the key.
+        """
+        global __KEY_VALID
+        return __KEY_VALID
+
```

Comment:
We can move this to the `init python in jn_utils` store above, no need to add another init block at the same level